firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/NestedMapEncoder.kt

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx.serialization
+
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.StructureKind
+import kotlinx.serialization.descriptors.elementNames
+import kotlinx.serialization.encoding.AbstractEncoder
+import kotlinx.serialization.encoding.CompositeEncoder
+import kotlinx.serialization.modules.EmptySerializersModule
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.serializer
+
+class NestedMapEncoder(
+    private var map: MutableMap<Int, MutableMap<String, Any?>> = mutableMapOf(),
+    private var depth: Int = 0,
+    private var list: MutableList<Any> = mutableListOf(),
```

Comment:
val instead of var

Reply:
I have changed all the var in this file into val now. There are only 4 not changed for `nest_depth` and `index` I must keep them as var